netplan/cli/commands/set.py

```diff
@@ -71,15 +73,55 @@ def command_set(self):
         with tempfile.TemporaryFile() as tmp:
             libnetplan.create_yaml_patch(yaml_path, value, tmp)
             tmp.flush()
+
             tmp.seek(0, io.SEEK_SET)
             parser.load_nullable_fields(tmp)
+            if filename:
+                # Validate the full YAML hierarchy without ignoring any NetDefs.
+                # Exit early if this YAML patch would lead to an invalid config.
+                # XXX: The origin file of each individual YAML setting/stanza
+                #      should be tracked individually, to avoid this
+                #      double-parsing workaround (LP: #2003727)
```

Comment:
thought (not-blocking): I'm still fairly convinced there are situations where the overall config passes validation but the overridden config doesn't, I just can't come up with one yet.

This bugs me. ;-)

Reply:
Right. I cannot rule that out either. If we can come up with any other failure case, let's document them in LP#2003727, so we can get them fixed properly.